### PR TITLE
Pol 571 moving average forecast

### PR DESCRIPTION
**Pull request by rshade**
### Description

[Describe what this change achieves]

### Issues Resolved

[List any existing issues this PR resolves]

### Contribution Check List

- [ ] All tests pass.
- [ ] New functionality includes testing.
- [ ] New functionality has been documented in the README if applicable
- [ ] New functionality has been documented in CHANGELOG.MD

### CODE REVIEW

File: cost/forecasting/moving_average_forecast/moving_average_forecast.pt [hunk 1]
```diff
@@ -0,0 +1,379 @@
...
+  forecasted_costs.push({ name: param_average_months + ' month average', average: simpleMovingAverage(arr_costs, param_average_months).reverse() });
+  size_difference = arr_costs.length - forecasted_costs[0]['average'].length
+  _.each(_.range(1, size_difference), function(item) {
+    forecasted_costs[0]['average'].push(forecasted_costs[0]['average'][forecasted_costs[0]['average'].length - 1])
+  })
+  var XAxisArray = _.range(1, arr_costs.length).reverse();
+  var chartData = 'chd=a:' + forecasted_costs[0]['average'].join(',') + '|' + arr_costs.reverse().join(',');
+  var chartXAxis = 'chxl=0:|' + XAxisArray.join('|');
+  forecasted_costs[0]['average'] = forecasted_costs[0]['average'].reverse()
+  var report = {
```
Comment (beagledeltasix) [inline cost/forecasting/moving_average_forecast/moving_average_forecast.pt:322]: Need to include encodeURI on each chart parameter for image-chart's API function. 320-333

File: cost/forecasting/moving_average_forecast/moving_average_forecast.pt [hunk 2]
```diff
@@ -0,0 +1,379 @@
...
+###############################################################################
+# Policy
+###############################################################################
+
+policy "policy_straight_line_forecast" do
+  validate $ds_forecast do
+    summary_template "{{parameters.param_average_months}} Month Moving Average"
+    detail_template <<-EOS
+# Moving Average Report for {{ rs_org_name }}
+![Moving Average](https://image-charts.com/chart?{{ data.chartType }}&{{ data.chartLabel }}&{{ data.chartSize }}&{{ data.chartTitle }}&{{ data.chartAxis }}&{{ data.chartXAxis }}&{{ data.chartAxisFormat }}&{{ data.chartData }}&{{ data.chartColors }}&{{ data.chartExtension }} "Forecasting Spend Chart")
```
Comment (beagledeltasix) [inline cost/forecasting/moving_average_forecast/moving_average_forecast.pt:359]: Need's the new API gateway url to interchange with image-charts in 359.

File: cost/forecasting/moving_average_forecast/CHANGELOG.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+# Changelog
+
+## v2.0
+
+- Initial Release
```
Comment (beagledeltasix) [inline cost/forecasting/moving_average_forecast/CHANGELOG.md:5]: What changes are being released / updated?

File: cost/forecasting/moving_average_forecast/straight_line_forecast.pt [hunk 4]
```diff
@@ -0,0 +1,380 @@
...
+      "Amortized Blended":"cost_amortized_blended_adj"
+    }
+    var billing_center_ids = []
+    if (param_billing_centers.length === 0){
+      var top_billing_centers = _.reject(ds_billing_centers, function(bc){ return bc.parent_id != null });
+      billing_center_ids = _.map(top_billing_centers, function(value, key){ return value.id });
+    } else {
+      // get array of billing center id's that match the names in param_billing_centers.
+      billing_center_names = _.map(param_billing_centers, function(name){ return name.toLowerCase(); });
+      billing_center_ids = _.compact(_.map(ds_billing_centers, function(value){ if(_.contains(billing_center_names, value.name.toLowerCase())){return value.id} }));
```
Comment (XOmniverse) [inline cost/forecasting/moving_average_forecast/straight_line_forecast.pt]: Might be a bit long for a one liner

File: cost/forecasting/moving_average_forecast/straight_line_forecast.pt [hunk 5]
```diff
@@ -0,0 +1,380 @@
...
+    }
+
+    var request = {
+      auth: "auth_rs",
+      verb: "POST",
+      host: "optima.rightscale.com",
```
Comment (XOmniverse) [inline cost/forecasting/moving_average_forecast/straight_line_forecast.pt]: Should this be hardcoded or a parameter? (Honest question)

**Review by beagledeltasix — DISMISSED** [review]

**Review by beagledeltasix — APPROVED** [review]

**Pull request merged**